src/masternodeconfig.cpp

```diff
@@ -16,41 +16,42 @@ bool CMasternodeConfig::read(boost::filesystem::path path) {
     if (!streamConfig.good()) {
         return true; // No masternode.conf file is OK
     }
-
+ 
     for(std::string line; std::getline(streamConfig, line); )
     {
         if(line.empty()) {
             continue;
         }
         std::istringstream iss(line);
-        std::string alias, ip, privKey, txHash, outputIndex;
-        iss.str(line);
-        iss.clear();
-        if (!(iss >> alias >> ip >> privKey >> txHash >> outputIndex)) {
-            LogPrintf("Could not parse masternode.conf line: %s\n", line.c_str());
-            streamConfig.close();
-            return false;
-        }
-
-        if(Params().NetworkID() == CChainParams::MAIN){
-            if(CService(ip).GetPort() != 17170) {
-                LogPrintf("Invalid port detected in masternode.conf: %s (must be 17170 for mainnet)\n", line.c_str());
+        std::string alias, ip, privKey, txHash, outputIndex, donation, donationAddress, donationPercent;
+        if (!(iss >> alias >> ip >> privKey >> txHash >> outputIndex >> donation)) {
+            donationAddress = "";
+            donationPercent = "";
+            iss.str(line);
+            iss.clear();
+            if (!(iss >> alias >> ip >> privKey >> txHash >> outputIndex)) {
+                LogPrintf("Could not parse masternode.conf line: %s\n", line.c_str());
+                streamConfig.close();
+                return false;
+            }
+        } else {
+            size_t pos = donation.find_first_of(":");
+            if(pos == string::npos) { // no ":" found
+                donationPercent = "100";
+                donationAddress = donation;
+            } else {
+                donationPercent = donation.substr(pos + 1);
+                donationAddress = donation.substr(0, pos);
+            }
+            CBitcoinAddress address(donationAddress);
+            if (!address.IsValid()) {
+                LogPrintf("Invalid Syndicate address in masternode.conf line: %s\n", line.c_str());
                 streamConfig.close();
                 return false;
             }
-        } else if(CService(ip).GetPort() == 17170) {
-            LogPrintf("Invalid port detected in masternode.conf: %s (17170 must be only on mainnet)\n", line.c_str());
-            streamConfig.close();
-            return false;
-        }
-
-        if (!(CService(ip).IsIPv4() && CService(ip).IsRoutable())) {
-            LogPrintf("Invalid Address detected in masternode.conf: %s (IPV4 ONLY) \n", line.c_str());
-            streamConfig.close();
-            return false;
```

Comment:
and lines 41-50 through here, which checks that the IP in the masternode.conf file is IPV4 and is routable.